activerecord/lib/active_record/connection_adapters/abstract/quoting.rb

```diff
@@ -158,8 +158,10 @@ def types_which_need_no_typecasting
 
       def _quote(value)
         case value
-        when String, ActiveSupport::Multibyte::Chars, Type::Binary::Data
+        when String, ActiveSupport::Multibyte::Chars
           "'#{quote_string(value.to_s)}'"
+        when Type::Binary::Data
+          "x'#{value.hex}'"
```

Comment:
This doesn't belong in the abstract adapter. It is not valid in the SQL standard.